crates/fluvio-controlplane-metadata/src/topic/deduplication.rs

```diff
@@ -0,0 +1,65 @@
+use std::{time::Duration, collections::BTreeMap};
+
+use derive_builder::Builder;
+use fluvio_protocol::{Encoder, Decoder};
+
+#[derive(Debug, Default, Builder, Clone, PartialEq, Eq, Encoder, Decoder)]
+#[cfg_attr(
+    feature = "use_serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "kebab-case")
+)]
+pub struct Deduplication {
+    pub bounds: Bounds,
+    pub filter: Filter,
+}
+
+#[derive(Debug, Default, Builder, Clone, PartialEq, Eq, Encoder, Decoder)]
+#[cfg_attr(
+    feature = "use_serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "kebab-case")
+)]
+pub struct Bounds {
+    #[cfg_attr(feature = "use_serde", serde(default, skip_serializing_if = "is_zero"))]
+    pub count: u64,
+    #[cfg_attr(
+        feature = "use_serde",
+        serde(
+            default,
+            skip_serializing_if = "Option::is_none",
+            with = "humantime_serde"
+        )
+    )]
+    pub age: Option<Duration>,
+}
```

Comment:
Not something to change for this PR, but I wonder if there is some commonality with Lookback logic we can reuse between the two (slightly different) uses as a generic "record (pre-?)selection"

Reply:
Although Bounds and Lookback are common now, they might differ later. Bound covers both preselection(Lookback) and limits for runtime stream processing (via SmartModule init params).